modules/nlu/src/backend/application/distributed-training-queue.ts

```diff
@@ -0,0 +1,53 @@
+import * as sdk from 'botpress/sdk'
+import * as NLU from 'common/nlu/engine'
+import { TrainingQueue, TrainingQueueOptions } from './training-queue'
+import { ITrainingRepository } from './training-repo'
+import { TrainingId, TrainerService, TrainingListener } from './typings'
+
+export class DistributedTrainingQueue extends TrainingQueue {
+  private _broadcastCancelTraining: (id: TrainingId) => Promise<void>
+  private _broadcastLoadModel: (botId: string, modelId: NLU.ModelId) => Promise<void>
+  private _broadcastRunTask: () => Promise<void>
+
+  constructor(
+    _trainingRepo: ITrainingRepository,
+    _errors: typeof NLU.errors,
+    _logger: sdk.Logger,
+    _trainerService: TrainerService,
+    private _distributed: typeof sdk.distributed,
+    _onChange: TrainingListener,
+    options: Partial<TrainingQueueOptions> = {}
+  ) {
+    super(_trainingRepo, _errors, _logger, _trainerService, _onChange, options)
+  }
+
+  public async initialize() {
+    await super.initialize()
+
+    const localCancelTraining = (trainId: TrainingId) => super.cancelTraining(trainId)
```

Comment:
Simply doing this
```ts
this._broadcastLoadModel = await this.broadcastFrom(super.cancelTraining.bind(this))
```
isn't working ?

Reply:
not sure, will have to check. `bp.distributed.broadcast` uses the function name so it must be defined